extensions/amp-smartlinks/0.1/test/test-amp-smartlinks.js

```diff
@@ -0,0 +1,265 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as DocumentReady from '../../../../src/document-ready';
+import * as LinkmateOptions from '../linkmate-options';
+import {AmpSmartlinks} from '../amp-smartlinks';
+import {LinkRewriterManager} from '../link-rewriter/link-rewriter-manager';
+import {SMARTLINKS_REWRITER_ID} from '../constants';
+import {Services} from '../../../../src/services';
+
+const helpersFactory = env => {
+  return {
+    createAmpSmartlinks(extensionAttrs) {
+      const ampTag = document.createElement('amp-smartlinks');
+
+      for (const attr in extensionAttrs) {
+        ampTag.setAttribute(attr, extensionAttrs[attr]);
+      }
+      ampTag.getAmpDoc = () => env.ampdoc;
+
+      return new AmpSmartlinks(ampTag);
+    },
+  };
+};
+
+describes.fakeWin(
+    'amp-smartlinks',
+    {
+      amp: {
+        extensions: ['amp-smartlinks'],
+      },
+    },
+    env => {
```

Comment:
def use parens in places like this

Reply:
![image](https://user-images.githubusercontent.com/9119148/51064039-baeab880-15cb-11e9-8b97-c9791e4e6457.png)
just so you'll believe me. AMP Eslint enforced

Reply:
😡amp
😄 phil